### PR TITLE
Upstream v3.20.0

### LINKED ISSUE #197: Can't upgrade tfbridge to v3.33.0 or higher due to azcore conflict
### What happened?

This is a follow-up from #192 for which a workaround was added in #196. However, we are currently blocked from taking a new version of the bridge because updating pulumi-terraform-bridge to v3.33.0 causes a build failure:

```
github.com/hashicorp/terraform-provider-vault/internal/provider
../../../go/pkg/mod/github.com/pulumi/terraform-provider-vault@v1.9.1-0.20221013032412-e762e637392b/internal/provider/auth_azure.go:185:9: tOpts.TenantID undefined (type policy.TokenRequestOptions has no field or method TenantID)
# github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:19:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:62:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:105:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:148:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:191:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:234:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:277:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:320:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:363:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:406:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:406:13: too many errors
```

The underlying issue seems to be because of these updates (going from tfbridge 3.32.0 to 3.33.0):
```
go: downloading github.com/pulumi/pulumi-terraform-bridge/v3 v3.33.0
go: upgraded cloud.google.com/go v0.102.1 => v0.103.0
go: upgraded github.com/Azure/azure-sdk-for-go v61.4.0+incompatible => v66.0.0+incompatible
go: upgraded github.com/Azure/azure-sdk-for-go/sdk/azcore v0.22.0 => v1.1.1
go: upgraded github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.13.2 => v1.0.0
go: upgraded github.com/Azure/azure-sdk-for-go/sdk/internal v0.9.1 => v1.0.0
go: upgraded github.com/Azure/go-autorest/autorest v0.11.24 => v0.11.28
go: upgraded github.com/Azure/go-autorest/autorest/adal v0.9.18 => v0.9.21
go: upgraded github.com/Azure/go-autorest/autorest/azure/cli v0.4.5 => v0.4.6
go: upgraded github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a => v0.0.0-20210307081110-f21760c49a8d
```

This is because of the bump of pulumi/pkg etc. to `v3.44.2` which brought in new versions of the above package in https://github.com/pulumi/pulumi/pull/10900/.

Vault terraform provider is still using the older versions of the above libraries which causes the conflict.

### Steps to reproduce

```
cd provider ;  go get -u github.com/pulumi/pulumi-terraform-bridge/v3; make build
```

### Expected Behavior

Build succeeds

### Actual Behavior

Build fails

### Output of `pulumi about`

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### LINKED ISSUE #297: Apply patch before v3.20.0
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

This might resolve #197. If not we'll need to use a forked provider and add the patch to that.

See https://github.com/pulumi/home/issues/3050#issuecomment-1689313059 for details of why.

### LINKED ISSUE #300: Upgrade terraform-provider-vault to v3.20.0
Release details: https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.20.0

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes #300
Fixes #297 

Update to upstream v3.20.0 with @lukehoban's patch to replace the `github.com/hashicorp/vault` dependency. 

Unfortunately, the dependency trim does not resolve (don't auto-resolve gh) #197.

### CODE REVIEW

File: provider/go.mod [hunk 1]
```diff
@@ -149,7 +139,6 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.19.0 // indirect
 	github.com/hashicorp/terraform-registry-address v0.2.1 // indirect
 	github.com/hashicorp/terraform-svchost v0.1.1 // indirect
-	github.com/hashicorp/vault v1.11.3 // indirect
```
Comment (iwahbe) [inline provider/go.mod:152]: We drop the vault dependency here.

File: provider/go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module github.com/pulumi/pulumi-vault/provider/v5
 go 1.21
 
 require (
-	github.com/hashicorp/terraform-provider-vault v1.9.1-0.20221006191932-d603953886dd
+	github.com/hashicorp/terraform-provider-vault v0.0.0
```
Comment (iwahbe) [inline provider/go.mod:6]: This is to make it obvious that we are using a `replace`, and not `v0.0.0`.

**Comment by iwahbe** [comment]
Snyk rejects on security but not license.

**Review by lukehoban — APPROVED** [review]
LGTM

**Review by t0yv0 — APPROVED** [review]

File: provider/go.mod [hunk 3]
```diff
@@ -206,10 +194,6 @@ require (
 	github.com/pkg/term v1.1.0 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/posener/complete v1.2.3 // indirect
-	github.com/prometheus/client_golang v1.11.1 // indirect
-	github.com/prometheus/client_model v0.3.0 // indirect
-	github.com/prometheus/common v0.30.0 // indirect
-	github.com/prometheus/procfs v0.7.3 // indirect
```
Comment (iwahbe) [inline provider/go.mod:212]: Synk failure looks spurious:

Screenshot 2023-09-08 at 10 08 20 AM

**Pull request merged**